src/common/snippets/include/snippets/kernel_executor_table.hpp

```diff
@@ -69,10 +132,37 @@ class KernelExecutorTable {
         OPENVINO_ASSERT(m_table.count(expr), "This expression doesn't have a registered kernel executor");
         return m_table.at(expr);
     }
+    /*** Updates every registered KernelExecutor in accordance with the corresponding expression */
+    void update_state() const {
+        for (const auto& record : m_table)
+            record.second->update_by_expression(record.first);
+    }
+
+    /*** Returns lambda function that contains current state of the table, and restores this state when called  */
+    std::function<void()> get_state_reset() {
+        auto current_state = get_state();
+        return [=]() { reset_state(current_state); };
+    }
```

Comment:
Honestly, this story with `state` looks not good for me. Let me please explain why:
This is problem only of CPU Plugin: we have several `SubgraphDynamicSpecialized` executor instances for each input shape set, some of them have the same JIT code (depends on broadcast mask). This JIT code addresses to shared pointers with kernel executors which are accessible for all `SubgraphDynamicSpecialized` executors (because it's sotred as shared pointer). It means that if we change kernel executors for one `SubgraphDynamicSpecialized` executor, it will update all other executors (for other input shapes). It happens because jit emitters points to table that is stored in shared `RuntimeConfig`.

The sense of `SubgraphDynamicSpecialized` executor is storing of input shape specific data: loop parameters, data offsets, buffer scratchpad size and offsets. Probably, we also should store kernels (which are specified by the corresponding input shapes) here and set them to `call_args` too?

Another idea is to turn down this `SubgraphDynamicSpecialized` executor and for each input shapes call `RuntimeConfigurator`. But it's very expensive activity (we saw it in Eltwise Subgraph case). Because of that we should provide safe dynamic kernel executors. For me, since specialized executors store runtime data (generated for target input shapes) and share JIT code between them, probably we should rewrite this code part (with kernel) as for other dynamic information: jit emitters get the data from config.

Also at the moment, `KernelExecutorTable` provides `state`-specific interface which is workaround needed only for CPU Plugin (because of executor architecture)

What do you think?